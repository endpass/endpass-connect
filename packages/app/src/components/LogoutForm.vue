<template>
  <form @submit.prevent="emitSubmit"> 
    <form-field>
      <message data-test="form-message">You are currently logged in. Press "Log out" button to terminate your current session.</message>
    </form-field>
    <form-controls>
      <v-button 
        :submit="true"
        :disabled="loading"
        type="primary"
        data-test="submit-button" 
      >Log out</v-button>
      <v-button 
        :disabled="!closable"
        data-test="cancel-button" 
        @click="emitCancel"
      >Close</v-button>
    </form-controls>
  </form>  
</template>

<script>
import VButton from './VButton.vue';
import Message from './Message.vue';
import FormField from './FormField.vue';
import FormControls from './FormControls.vue';

export default {
  name: 'LogoutForm',

  props: {
    closable: {
      type: Boolean,
      default: true,
    },

    loading: {
      type: Boolean,
      default: false,
    },
  },

  methods: {
    emitSubmit() {
      this.$emit('submit');
    },

    emitCancel() {
      if (this.closable) {
        this.$emit('cancel');
      }
    },
  },

  components: {
    VButton,
    Message,
    FormField,
    FormControls,
  },
};
</script>
