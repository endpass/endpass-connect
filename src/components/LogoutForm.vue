<template>
  <form @submit.prevent="emitSubmit"> 
    <form-field>
      <message data-test="form-message">You are currently logged in. Press "Log out" button to terminate your current session.</message>
    </form-field>
    <form-controls>
      <v-button 
        :submit="true"
        :disabled="loading"
        type="primary"
        data-test="submit-button" 
      >{{ primaryButtonLabel }}</v-button>
      <v-button 
        :disabled="!closable || loading"
        data-test="cancel-button" 
        @click="emitCancel"
      >Close</v-button>
    </form-controls>
  </form>  
</template>

<script>
import VButton from './VButton.vue';
import Message from './Message.vue';
import FormField from './FormField.vue';
import FormControls from './FormControls.vue';

export default {
  name: 'LogoutForm',

  props: {
    closable: {
      type: Boolean,
      default: true,
    },

    loading: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    primaryButtonLabel() {
      return !this.loading ? 'Log out' : 'Loading...';
    },
  },

  methods: {
    emitSubmit() {
      if (!this.loading) {
        this.$emit('submit');
      }
    },

    emitCancel() {
      if (this.closable) {
        this.$emit('cancel');
      }
    },
  },

  components: {
    VButton,
    Message,
    FormField,
    FormControls,
  },
};
</script>
