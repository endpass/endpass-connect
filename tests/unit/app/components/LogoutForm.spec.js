import { shallowMount, mount } from '@vue/test-utils';
import LogoutForm from '@/components/LogoutForm.vue';

describe('LogoutForm', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(LogoutForm, {
        propsData: {},
      });
    });

    it('should correctly render LogoutForm component', () => {
      expect(wrapper.name()).toBe('LogoutForm');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(LogoutForm, {
        propsData: {},
      });
    });

    describe('submit feature', () => {
      it('should emit submit on click submit button by default', () => {
        wrapper.find('form').trigger('submit');

        expect(wrapper.emitted().submit).toBeTruthy();
      });

      it('should not emit submit on click submit button if it is loading', () => {
        wrapper.setProps({
          loading: true,
        });

        wrapper.find('form').trigger('submit');

        expect(wrapper.emitted().submit).toBeFalsy();
      });
    });

    describe('cancel feature', () => {
      it('should emit close on click cancel button by default', () => {
        wrapper.find('[data-test=cancel-button]').trigger('click');

        expect(wrapper.emitted().cancel).toBeTruthy();
      });

      it('should not emit close on click cancel button if closable is false', () => {
        wrapper.setProps({
          closable: false,
        });

        wrapper.find('[data-test=cancel-button]').trigger('click');

        expect(wrapper.emitted().cancel).toBeFalsy();
      });
    });
  });
});
