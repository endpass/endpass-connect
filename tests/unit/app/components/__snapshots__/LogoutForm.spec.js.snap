// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutForm render should correctly render LogoutForm component 1`] = `
<form>
  <form-field-stub>
    <message-stub data-test="form-message">You are currently logged in. Press "Log out" button to terminate your current session.</message-stub>
  </form-field-stub>
  <form-controls-stub>
    <v-button-stub type="primary" submit="true" data-test="submit-button">Log out</v-button-stub>
    <v-button-stub data-test="cancel-button">Close</v-button-stub>
  </form-controls-stub>
</form>
`;
